_dev/src/ts/api/SegmentApi.ts

```diff
@@ -42,17 +49,21 @@ class Analytics {
    * @description Tracks an event with optional properties.
    */
   public track = async (event: string, properties?: object) => {
-    await this.analytics.track(
-      event,
-      {
-        module: 'autoupgrade',
-        ...this.#getDefaultProperties(),
-        ...properties
-      },
-      {
-        page: this.#getMaskedPageData()
-      }
-    );
+    try {
+      await this.analytics.track(
+        event,
+        {
+          module: 'autoupgrade',
+          ...this.#getDefaultProperties(),
+          ...properties
+        },
+        {
+          page: this.#getMaskedPageData()
+        }
+      );
+    } catch (error) {
+      console.error(`[Analytics] Failed to track event "${event}".`, error);
+    }
```

Comment:
Let's avoid spamming the browser console if we can't send an event.
We can check this.analytics is defined before calling it `this.analytics?.`

```suggestion
      await this.analytics?.track(
        event,
        {
          module: 'autoupgrade',
          ...this.#getDefaultProperties(),
          ...properties
        },
        {
          page: this.#getMaskedPageData()
        }
      );
```